portal.go

```diff
@@ -1725,6 +1690,17 @@ func (portal *Portal) convertMediaMessage(intent *appservice.IntentAPI, source *
 		}
 	}
 
+	var width, height int
+	messageWithDimensions, ok := msg.(MediaMessageWithDimensions)
+	if ok {
+		width = int(messageWithDimensions.GetWidth())
+		height = int(messageWithDimensions.GetHeight())
+	}
+	if width == 0 && height == 0 && strings.HasPrefix(msg.GetMimetype(), "image/") {
+		cfg, _, _ := image.DecodeConfig(bytes.NewReader(data))
+		width, height = cfg.Width, cfg.Height
+	}
+
```

Comment:
I'm pretty sure this block shouldn't be moved down here, decoding the image requires the raw data (pre-`encryptFile`)